pheps/phep-0003.md

```diff
@@ -0,0 +1,374 @@
+There is no expectation that a package "deprecate" an older dependency before dropping support for it. 
+However, there is an expectation that maximum or exact requirements (e.g., `numpy<2` or `matplotlib==3.5.3`) be set only when absolutely necessary (and that issues be immediately created to remove such requirements), and packages must not require versions of any dependency older than 24 months. 
+Additionally, if a package has been supporting specific OS versions and CPU architectures (e.g., releasing binary [wheels](https://packaging.python.org/en/latest/discussions/package-formats/#what-is-a-wheel)), this support should continue for new OS versions and architectures to maintain the same level of support as before.
+
+This new policy will replace the current standard [#11](https://github.com/heliophysicsPy/standards/blob/main/standards.md#standards) in the PyHC standards document with the following new text:
+
+> **11. Python and Upstream Package Support:** All packages should support minor Python versions released within the last 36 months (3 years) and upstream core Scientific Python packages released within the last 24 months (2 years). 
+Additionally, packages should support new versions within 6 months of their release (see [PHEP 3](https://github.com/heliophysicsPy/standards/pull/29)). 
+
+Lastly, if there is a Python 4 or other significant changes in dependencies, this policy will have to be reviewed in light of the community's and projects' best interests.
+
+# Backwards Compatibility
+<a name="backwards-compatibility"></a>
+This policy potentially introduces backwards incompatibilities by enforcing a new support timeline, which may encourage some packages to drop support for older dependency versions sooner than planned.
+
+# Security Implications
+<a name="security-implications"></a>
+There are no direct security implications of this policy. 
+However, ensuring packages are updated to newer dependency versions may improve security by incorporating fixes and improvements from newer releases.
+
+# How to Teach This
+<a name="how-to-teach-this"></a>
+ - A new web page on the PyHC website will detail the support policy and include a graphical timeline of the schedule (similar to the Gantt chart above).
+ - Automated email reminders will be sent via the PyHC mailing list quarterly and near important drop/support dates to remind package maintainers of the schedule.
+
+# Reference Implementation
+<a name="reference-implementation"></a>
+Multiple PyHC packages already follow this version support policy. 
+One notable example is PlasmaPy which currently [documents their SPEC 0-based policy](https://docs.plasmapy.org/en/stable/contributing/coding_guide.html#python-and-dependency-version-support) and even mentions it in comments inside their [pyproject.toml](https://github.com/PlasmaPy/PlasmaPy/blob/main/pyproject.toml) file.
+
+## Code to generate support and drop schedules:
+```python
```

Comment:
Can it be explicitly noted that this code is the source of the included Gantt chart? Does it make more sense to put it elsewhere (in this repository or otherwise) and link it from here, as something that is live updated as necessary without having to update the PHEP?

Reply:
You got it. I'll add "`The following code can be used to generate support and drop schedules, including the Gantt chart above.`" between the header and the code.

I considered putting the code elsewhere, like in a separate file and linking to it, but decided not to for two main reasons: (1) SPEC 0 included their code in-line inside the SPEC and I liked that. (2) I honestly do not intend to formally maintain this code. Sure the dates etc will eventually become obsolete after enough time passes, but ain't nobody got time to remember to update dates in an obscure script! It's valid now and a good enough starting point for anyone who wants to use it in the future (likely only me tbh).